### PR TITLE
function_testing.py docstring update

### LINKED ISSUE #21519: [Documentation]: Helpers.test_function docstring require backend_to_test keyword addition
### Path

https://github.com/unifyai/ivy/blob/master/ivy_tests/test_ivy/helpers/function_testing.py#L88

### Section

do not found in section

### Actual Behavior

`helpers.test_function` has keyword argument `backend_to_test` but is not mentioned in the docstring. 

how it is right now

```python
def test_function(
    *,
    input_dtypes: Union[ivy.Dtype, List[ivy.Dtype]],
    test_flags: FunctionTestFlags,
    fn_name: str,
    rtol_: float = None,
    atol_: float = 1e-06,
    test_values: bool = True,
    xs_grad_idxs=None,
    ret_grad_idxs=None,
    backend_to_test: str,
    on_device: str,
    return_flat_np_arrays: bool = False,
    **all_as_kwargs_np,
):
    """
    Test a function that consumes (or returns) arrays for the current backend by
    comparing the result with numpy.

    Parameters
    ----------
    input_dtypes
        data types of the input arguments in order.
    test_flags
        FunctionTestFlags object that stores all testing flags, including:
        num_positional_args, with_out, instance_method, as_variable,
        native_arrays, container, gradient
    fw
        current backend (framework).
    fn_name
        name of the function to test.
    rtol_
        relative tolerance value.
    atol_
        absolute tolerance value.
    test_values
        if True, test for the correctness of the resulting values.
    xs_grad_idxs
        Indices of the input arrays to compute gradients with respect to. If None,
        gradients are returned with respect to all input arrays. (Default value = None)
    ret_grad_idxs
        Indices of the returned arrays for which to return computed gradients. If None,
        gradients are returned for all returned arrays. (Default value = None)
    on_device
        The device on which to create arrays
    return_flat_np_arrays
        If test_values is False, this flag dictates whether the original returns are
        returned, or whether the flattened numpy arrays are returned.
    all_as_kwargs_np
        input arguments to the function as keyword arguments.

    Returns
    -------
    ret
        optional, return value from the function
    ret_gt
        optional, return value from the Ground Truth function
  """
  
 ```

### Expected Behavior

how it should be,

```python
def test_function(
    *,
    input_dtypes: Union[ivy.Dtype, List[ivy.Dtype]],
    test_flags: FunctionTestFlags,
    fn_name: str,
    rtol_: float = None,
    atol_: float = 1e-06,
    test_values: bool = True,
    xs_grad_idxs=None,
    ret_grad_idxs=None,
    backend_to_test: str,
    on_device: str,
    return_flat_np_arrays: bool = False,
    **all_as_kwargs_np,
):
    """
    Test a function that consumes (or returns) arrays for the current backend by
    comparing the result with numpy.

    Parameters
    ----------
    input_dtypes
        data types of the input arguments in order.
    test_flags
        FunctionTestFlags object that stores all testing flags, including:
        num_positional_args, with_out, instance_method, as_variable,
        native_arrays, container, gradient
    fw
        current backend (framework).
    fn_name
        name of the function to test.
    rtol_
        relative tolerance value.
    atol_
        absolute tolerance value.
    test_values
        if True, test for the correctness of the resulting values.
    xs_grad_idxs
        Indices of the input arrays to compute gradients with respect to. If None,
        gradients are returned with respect to all input arrays. (Default value = None)
    ret_grad_idxs
        Indices of the returned arrays for which to return computed gradients. If None,
        gradients are returned for all returned arrays. (Default value = None)
    on_device
        The device on which to create arrays
    return_flat_np_arrays
        If test_values is False, this flag dictates whether the original returns are
        returned, or whether the flattened numpy arrays are returned.
    all_as_kwargs_np
        input arguments to the function as keyword arguments.
    backend_to_test
        set the backend to test(tentative)

    Returns
    -------
    ret
        optional, return value from the function
    ret_gt
        optional, return value from the Ground Truth function
    """
```

### Screenshots

image

### LINKED ISSUE #21911: [Documentation]: ivy_tests/test_ivy/helpers/function_testing.py doc string addition to the functions
### Path

https://github.com/unifyai/ivy/blob/master/ivy_tests/test_ivy/helpers/function_testing.py#L88

### Section

do not found in section

### Actual Behavior

-

### Expected Behavior

Addition of the dosdtring to the function that do not contain the doc string

### Screenshots

-

### PR DESCRIPTION
**Pull request by rajveer43**
close #21519
close #21911

### CODE REVIEW

**Comment by theRealBird** [comment]
Hi @rajveer43,
Thank you for your contribution to Ivy. I see that you have started adding docstrings for other functions as well. We appreciate and celebrate your efforts! However, in order for us to keep track of what has been implemented/ what's missing could you please open relevant issues for all of the docstrings which you are adding?

Your changes look good to me! But I will let @CatB1t, take a look once too as the original issue has been assigned to him. Thanks!

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 1]
```diff
@@ -125,6 +125,9 @@ def test_function(
         returned, or whether the flattened numpy arrays are returned.
     all_as_kwargs_np
         input arguments to the function as keyword arguments.
+    backend_to_test
+        The backend (framework) to test the function against, e.g., "torch", "numpy",
+        etc.
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:130]:
```suggestion
      The backend framework to be tested.
```

Let's not include examples in the docstring for parameter. Let's update the other docstring similarly, Thanks!

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 2]
```diff
@@ -860,6 +866,56 @@ def gradient_test(
...
+    backend and a ground truth backend, and compare the gradients.
+
+    Parameters
+    ----------
+    fn
+        The function to test. If a string, it is assumed to be the name of the
+        function. If callable, it is the actual function to be tested.
+    all_as_kwargs_np
+        Input arguments to the function as keyword arguments.
+    args_np : tuple
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:880]: let's omit the type of the argument

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 3]
```diff
@@ -860,6 +866,56 @@ def gradient_test(
...
+        Tuple containing the numpy arrays of the positional arguments.
+    kwargs_np
+        Tuple containing the numpy arrays of the keyword arguments.
+    input_dtypes
+        Data types of the input arguments in order.
+    test_flags
+        Frontend function flags object that stores all testing flags.
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:887]:
```suggestion
        Function Test Flags object that contains test flags to be applied.
```

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 4]
```diff
@@ -860,6 +866,56 @@ def gradient_test(
...
+    test_compile
+        If True, test compilation. (Default value: False)
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:889]: Let's omit default values.

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 5]
```diff
@@ -860,6 +866,56 @@ def gradient_test(
...
+    rtol_
+        Relative tolerance value for gradient comparison. (Default value: None)
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:891]: Let's omit default values.

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 6]
```diff
@@ -860,6 +866,56 @@ def gradient_test(
...
+    atol_
+        Absolute tolerance value for gradient comparison. (Default value: 1e-06)
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:893]: Let's omit default values.

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 7]
```diff
@@ -860,6 +866,56 @@ def gradient_test(
...
+        Indices of the returned arrays for which to return computed gradients.
+        If None, gradients are returned for all returned arrays.
+        (Default value: None)
+    backend_to_test
+        The backend (framework) to test the function against,
+        e.g., "torch", "numpy", etc.
+    ground_truth_backend
+        The backend (framework) used as ground truth for gradient comparison.
+    on_device
+        The device on which to create arrays.
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:908]:
```suggestion
        The device on which to test the function on.
```

File: ivy_tests/test_ivy/helpers/function_testing.py [hunk 8]
```diff
@@ -860,6 +866,56 @@ def gradient_test(
...
+
+    Returns
+    -------
+    None
```
Comment (CatB1t) [inline ivy_tests/test_ivy/helpers/function_testing.py:912]: We can remove this.

**Review by CatB1t — CHANGES_REQUESTED** [review]

**Comment by KareemMAX** [comment]
Hey @rajveer43,
I just checked on this PR. Maybe @sherry30 can take @CatB1t's place. If you don't mind @CatB1t ofc 😅

**Pull request closed without merging**